File: problems/118-pascals-triangle/solution.c
int** generate(int num_rows, int* return_size, int** return_column_sizes)
{
    int **result = (int**)malloc(num_rows * sizeof(int*));
    *return_column_sizes = (int*)malloc(num_rows * sizeof(int));
    *return_size = num_rows;

    for (int i = 0; i < num_rows; i++) {
        result[i] = (int*)malloc((i + 1) * sizeof(int));
        (*return_column_sizes)[i] = i + 1;

        result[i][0] = 1;
        result[i][i] = 1;
        
        for (int j = 1; j < i; j++) {
            result[i][j] = result[i - 1][j] + result[i - 1][j - 1];
        }
    }
    return result;
}

File: problems/118-pascals-triangle/solution.py
class Solution:
    def generate(self, num_rows: int) -> list[list[int]]:
        result = [[1]]

        for _ in range(num_rows - 1):
            new_row = result[-1] + [1]
            new_row_copy = new_row.copy()
            for i in range(1, len(new_row) - 1):
                new_row[i] += new_row_copy[i - 1]
            result.append(new_row)
        return result
        